maplibre/src/coords.rs

```diff
@@ -481,15 +538,15 @@ impl ViewRegion {
     pub fn iter(&self) -> impl Iterator<Item = WorldTileCoords> + '_ {
         (self.min_tile.x - self.padding..self.max_tile.x + 1 + self.padding).flat_map(move |x| {
             (self.min_tile.y - self.padding..self.max_tile.y + 1 + self.padding).map(move |y| {
-                let tile_coord: WorldTileCoords = (x, y, self.z as u8).into();
+                let tile_coord: WorldTileCoords = (x, y, self.z.0 as u8).into();
```

Comment:
We should construct a WorldTileCoords directly from the ZoomLevel. We should no unpack if first and then create a new one.

Maybe implement the From `(u32, u32, ZoomLevel)` trait for WorldTileCoords